packages/store/src/internal/lifecycle-state-manager.ts

```diff
@@ -25,33 +25,51 @@ export class LifecycleStateManager {
         mergeMap(() => this.bootstrapper.appBootstrapped$),
         filter(appBootstrapped => !!appBootstrapped)
       )
-      .subscribe(() => {
-        this.invokeBootstrap(results!.states);
-      });
+      .subscribe(() => this.invokeBootstrap(results!.states));
   }
 
   /**
    * Invoke the init function on the states.
    */
   invokeInit(stateMetadatas: MappedStore[]): void {
-    this.invokeLifecycleHooks(stateMetadatas, LifecycleHooks.NgxsOnInit);
+    for (const metadata of stateMetadatas) {
+      const instance: NgxsLifeCycle = metadata.instance;
+
+      if (instance.ngxsOnChanges) {
```

Comment:
Can we safely assume that instance is not null here?

Reply:
Yes

![image](https://user-images.githubusercontent.com/12021443/67335134-ea9bf900-f52b-11e9-971f-fc57f85b436b.png)

Reply:
👍